### PR TITLE
Added boolean operator and more

### PR DESCRIPTION
**Pull request by neroroxxx**
Added a bool operator to check if a midi device is connected, added methods to get vid and pid to match those of Teensy's 3.6 built-in USB Host.

### CODE REVIEW

**Comment by Lauszus** [comment]
@YuuichiAkagawa can you take a look at this?

**Comment by YuuichiAkagawa** [comment]
@Lauszus
Sure.

**Comment by YuuichiAkagawa** [comment]
@neroroxxx 
Why do you want to use operator overloading?
Other drivers are not using it.

**Comment by neroroxxx** [comment]
@YuuichiAkagawa it just seems like a clean way to use it, you can check if something is connected to the host AND if the device is a midi device in one statement.

Usb.Task();
if(Midi){
pollMIDI();
}

**Comment by YuuichiAkagawa** [comment]
@neroroxxx But why only MIDI?

This code is pick from Teensy's library.
https://github.com/PaulStoffregen/USBHost_t36/blob/master/USBHost_t36.h#L388-L399

@Lauszus
Which do you think is better for applying to the whole UHS2 or only MIDI?

**Comment by Lauszus** [comment]
@YuuichiAkagawa I don't see the benefit. I have only done it for the SPP class: https://github.com/felis/USB_Host_Shield_2.0/blob/master/SPP.h#L80-L82, as I wanted it to act much like the Serial class in Arduino.

File: usbh_midi.h [hunk 1]
```diff
@@ -66,6 +66,11 @@ class USBH_MIDI : public USBDeviceConfig
 public:
         uint16_t pid, vid;
         USBH_MIDI(USB *p);
+        operator bool(){
+          return (pUsb->getUsbTaskState()==USB_STATE_RUNNING && isMidiFound);
+        }
+        uint16_t idVendor(){return vid;}
+        uint16_t idProduct(){return pid;}
```
Comment (Lauszus) [inline usbh_midi.h]: @neroroxxx great if you could add some whitespace and use 8 spaces before the return statement, so it follows the coding style of the existing code.

**Comment by YuuichiAkagawa** [comment]
@Lauszus OK. Please merge when the correction is completed.

**Comment by Lauszus** [comment]
Merged manually in https://github.com/felis/USB_Host_Shield_2.0/commit/8ef3db254dcfa1fd27588c8401ac10cafe6c2bf4. Thanks for your contribution!

**Pull request closed without merging**